sql/catalyst/src/main/scala/org/apache/spark/sql/types/Decimal.scala

```diff
@@ -286,6 +288,9 @@ object Decimal {
   /** Maximum number of decimal digits a Long can represent */
   val MAX_LONG_DIGITS = 18
 
+  /** Maximum precision a Decimal can support */
+  val MAX_PRECISION = 38
```

Comment:
ok. we can use `MathContext.UNLIMITED` for unlimited precision on scala decimal.